lib/pleaserun/detector.rb

```diff
@@ -25,14 +25,19 @@ def detect_platform
 
   def detect_systemd
     # Expect a certain directory
-    return false unless File.directory?("/lib/systemd/system")
+    return false unless File.directory?("/lib/systemd/system") || File.directory?("/usr/lib/systemd/system")
+
+    # Verify presence of systemctl before proceeding.  This catches an edge
+    # case in ubuntu 12.04 where the systemctl path exists, but no binary
+    out, status = execute([ "which", "systemctl" ])
```

Comment:
I'm fine with that.  It just needs to return non-zero in the event it's not found. for the original code here to work.